client/softlayer_client.go

```diff
@@ -22,33 +22,34 @@ const (
 	SL_GO_NON_VERBOSE  = "SL_GO_NON_VERBOSE"
 )
 
-type softLayerClient struct {
+type SoftLayerClient struct {
 	username string
 	apiKey   string
 
 	templatePath string
 
-	httpClient *http.Client
+	// HTTPClient is used for HTTP requests. By default DefaultHTTPClient is
+	// being used.
+	HTTPClient *http.Client
 
 	softLayerServices map[string]softlayer.Service
 
 	nonVerbose bool
 }
 
-func NewSoftLayerClient(username, apiKey string) *softLayerClient {
-	pwd, _ := os.Getwd()
-	slc := &softLayerClient{
+func NewSoftLayerClient(username, apiKey string) *SoftLayerClient {
+	pwd, err := os.Getwd()
+	if err != nil {
+		panic(err) // this should be handled by the user
```

Comment:
Actually I think we can remove the call for `pwd, err := os.Getwd()` since it's used to set `templatePath` which is only used in `GenerateRequestBody` and I think that's not used much anywhere.

I have to double check. So leave the `panic(err)` and I will open an issue to investigate removing `templatePath` and `GenerateRequestBody` from interface. Do not use them since they will likely be removed.

Thanks.